packages/ui/src/components/va-carousel/VaCarousel.stories.ts

```diff
@@ -1,12 +1,203 @@
+  })
+}
+
+export const Stateful = () => ({
+  components: { VaCarousel },
+  ...carouselSetup,
+  template: `
+    [true] - should slide
+    <VaCarousel :items="items" stateful />
+    [false] - should not slide
+    <VaCarousel :items="items" />
+  `,
+})
+
+Stateful.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const carousel = document.querySelectorAll('div.va-carousel__slide')
+  const [statefulslideTwo, slideTwo] = canvas.getAllByRole('button', { name: 'go slide 2' }) as HTMLElement[]
+
+  await step('Should head to slide 2', async () => {
+    userEvent.click(statefulslideTwo)
+    await sleep()
+    expect(carousel[1].getAttribute('aria-current')).toEqual('true')
+  })
+
+  await step('Should do nothing', async () => {
+    userEvent.click(slideTwo)
+    await sleep()
+    expect(carousel[6].getAttribute('aria-current')).toEqual('false')
+  })
+}
+
+export const Vertical = () => ({
+  components: { VaCarousel, VaSwitch },
+  ...carouselSetup,
+  template: `
+    <VaCarousel :items="items" v-model="currentSlide" vertical />
+  `,
+})
+
+export const Arrows = () => ({
+  components: { VaCarousel, VaSwitch },
+  ...carouselSetup,
+  template: `
+    <va-switch
+      v-model="areArrowsVisible"
+      class="mb-6"
+      label="Arrows"
+      size="small"
+    />
+    <VaCarousel :items="items" v-model="currentSlide" :arrows="areArrowsVisible" />
+  `,
+})
+
+export const Indicators = () => ({
+  components: { VaCarousel, VaSwitch },
+  ...carouselSetup,
+  template: `
+    <div class="space-x-2 mb-6">
+      <va-switch
+        v-model="areIndicatorsVisible"
+        label="Indicators"
+        size="small"
+      />
+      <va-switch
+        v-model="triggerOnHover"
+        label="Trigger on hover"
+        size="small"
+      />
+    </div>
+    <VaCarousel :items="items" v-model="currentSlide" :indicators="areIndicatorsVisible" :indicatorTrigger="triggerOnHover ? 'hover' : 'click'" />
+  `,
+})
+
+export const Infinite = () => ({
+  components: { VaCarousel, VaSwitch },
+  ...carouselSetup,
+  template: `
+    <VaCarousel :items="items" v-model="currentSlide" infinite />
+  `,
+})
+
+Infinite.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const carousel = document.querySelectorAll('div.va-carousel__slide')
+  const slideFive = canvas.getByRole('button', { name: 'go slide 5' }) as HTMLElement
+  const goForward = canvas.getByRole('button', { name: 'go next slide' }) as HTMLElement
+
+  await step('Should head to slide 1 from slide 5', async () => {
+    userEvent.click(slideFive)
+    await sleep()
+    userEvent.click(goForward)
+    await sleep()
+    expect(carousel[0].getAttribute('aria-current')).toEqual('true')
+  })
 }
 
-export const Default = defineComponent({
+export const Autoscroll = () => ({
+  components: { VaCarousel, VaSwitch },
+  ...carouselSetup,
+  template: `
+    [default]
+    <VaCarousel :items="items" stateful autoscroll />
```

Comment:
Let's have only one example here.

Autoscroll should behave as infinite always.

There is no case, where we need to go:
1,2,3,2,1.

We always need:
1,2,3,1

You can create a bug report or fix it on your own.